### PR TITLE
LPS-194181 | Master

### PR DESCRIPTION
**Pull request by davisaints**
**JIRA Ticket: [LPS-194181](https://liferay.atlassian.net/browse/LPS-194181)**

### CODE REVIEW

**Comment by davisaints** [comment]
ci:test:sf

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro [hunk 1]
```diff
@@ -916,6 +916,8 @@ definition {
 			value1 = ${selectOption});
 
 		if ((${fieldName} == "Choose a Data Provider") || (${fieldName} == "Output Parameter")) {
+			WaitForElementPresent(locator1 = "FormFields#SELECT_FIELD_DATA_PROVIDER_LIST");
+
```
Comment (patriciajeanperez) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:920]: @davisaints do we need this?

**Review by patriciajeanperez — CHANGES_REQUESTED** [review]
@davisaints Let's make the test run faster by making use of existing JSON macros. Thanks!

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase [hunk 2]
```diff
@@ -640,6 +640,213 @@ definition {
...
+				category = "Site Builder",
+				portlet = "Pages");
+
+			PagesAdmin.addPage(pageName = "Form Page");
+
+			Navigator.gotoPage(pageName = "Form Page");
+
+			Portlet.addPG(
+				pageName = "Form Page",
+				portletName = "Form");
```
[the remark this answers is not in the document]
Comment (davisaints) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase:790]: @patriciajeanperez yes, but not all this block of steps. I can just add the layout via json

**Comment by davisaints** [comment]
@patriciajeanperez please see my comments. The new index has been attached

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro (hunk 1 above)
Comment (patriciajeanperez) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:920]: @davisaints also on CI?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase (hunk 2 above)
[the remark this answers is not in the document]
Comment (davisaints) [reply portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/fields/FormsSelectFromListField.testcase]: @patriciajeanperez hm, It's true! Sorry for the disbelief lol

File: portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro (hunk 1 above)
Comment (davisaints) [reply portal-web/test/functional/com/liferay/portalweb/macros/FormFields.macro:920]: No, only locally. For now I will remove this, if there is a problem with this I will quickly correct it

**Comment by davisaints** [comment]
@patriciajeanperez looks good now

**Comment by patriciajeanperez** [comment]
ci:forward

**Pull request closed without merging**